src/main/flight/gps_rescue.c

```diff
@@ -293,8 +293,8 @@ void rescueStop()
 // Things that need to run regardless of GPS rescue mode being enabled or not
 void idleTasks()
 {
-    // Do not calculate any of the idle task values when we are not flying
-    if (!ARMING_FLAG(ARMED)) {
+    // Do not calculate any of the idle task values when we are not flying.  No sense in storing max altitude or throttle samples if not climbing.
```

Comment:
How about this:
1. Introduce a small info-function in position.c:
bool getFlagAltitudeOffsetSet(void)
{
    return altitudeOffsetSet;
}
2. Read the flag in gps_rescue.c:
    if (!ARMING_FLAG(ARMED) || !getFlagAltitudeOffsetSet()) {
        return;
    }
IMO this would assure that the "non-offsetted" altitude isn't read until the offset got determined.

Reply:
@AirBreak69: That should work.